core/src/test/kotlin/org/stellar/anchor/sep6/Sep6ServiceTest.kt

```diff
@@ -172,6 +172,36 @@ class Sep6ServiceTest {
     )
   }
 
+  @Test
+  fun `test deposit with funding_method`() {
+    val slotTxn = slot<Sep6Transaction>()
+    every { txnStore.save(capture(slotTxn)) } returns null
+
+    val slotEvent = slot<AnchorEvent>()
+    every { eventSession.publish(capture(slotEvent)) } returns Unit
+
+    val fundingMethod = "SWIFT"
+    val request =
+      StartDepositRequest.builder()
+        .assetCode(TEST_ASSET)
+        .account(TEST_ACCOUNT)
+        .fundingMethod(fundingMethod)
+        .amount("100")
+        .build()
+    sep6Service.deposit(token, request)
+
+    // Verify validations
+    verify(exactly = 1) { requestValidator.getDepositAsset(TEST_ASSET) }
+    verify(exactly = 1) { requestValidator.validateAccount(TEST_ACCOUNT) }
+
+    // Verify effects
+    verify(exactly = 1) { txnStore.save(any()) }
```

Comment:
Can we add an assertion on the transaction saved to the database? Updating all the tests in this class may be easier, replacing `type` with `funding_method` and adding a few additional tests for just `type`. WDYT?

Reply:
This seems to complicate things unnecessarily. If the goal is simply to test whether `funding_method` or `type` are processed correctly, I can extract the new logic into a helper function and test from there.

Reply:
We should at least have a test that `funding_method` is persisted in the database. Whether that's done by refactoring into a helper method or replacing all the `.type` with `.fundingMethod`, I'll leave it to you. In my opinion, replacing all the `type` calls would be easier.